packages/contracts-bedrock/src/L2/L1Block.sol

```diff
@@ -73,4 +85,35 @@ contract L1Block is ISemver {
         l1FeeOverhead = _l1FeeOverhead;
         l1FeeScalar = _l1FeeScalar;
     }
+
+    /// @notice Updates the L1 block values for an Ecotone upgraded chain.
+    /// Params are packed and passed in as raw msg.data instead of ABI to reduce calldata size.
+    /// Params are expected to be in the following order:
+    ///   1. _basefeeScalar      L1 basefee scalar
+    ///   2. _blobBasefeeScalar  L1 blob basefee scalar
+    ///   3. _sequenceNumber     Number of L2 blocks since epoch start.
+    ///   4. _timestamp          L1 timestamp.
+    ///   5. _number             L1 blocknumber.
+    ///   6. _basefee            L1 basefee.
+    ///   7. _blobBasefee        L1 blobBasefee.
+    ///   8. _hash               L1 blockhash.
+    ///   9. _batcherHash        Versioned hash to authenticate batcher by.
+    function setL1BlockValuesEcotone() external {
+        assembly {
+            // Revert if the caller is not the depositor account.
+            if xor(caller(), DEPOSITOR_ACCOUNT) {
+                mstore(0x00, 0x3cc50b45) // 0x3cc50b45 is the 4-byte selector of "NotDepositor()"
+                revert(0x1C, 0x04) // returns the stored 4-byte selector from above
+            }
+            let data := calldataload(4)
+            // sequencenum (uint64), blobBasefeeScalar (uint32), basefeeScalar (uint32)
+            sstore(sequenceNumber.slot, shr(128, calldataload(4)))
+            // number (uint64) and timestamp (uint64)
+            sstore(number.slot, shr(128, calldataload(20)))
```

Comment:
Can we please add the code that cleaned the dirty bits back?

Reply:
```suggestion
                        sstore(sequenceNumber.slot, and(shr(128, data), 0xFFFFFFFFFFFFFFFF)) // uint64
```

Reply:
See https://github.com/ethereum-optimism/optimism/pull/8726#discussion_r1443262501

Reply:
nvm - the encoding of the calldata changed since I last reviewed. Disregard.

Reply:
Just realized there are a few more bits that should be empty that I should test for, pls hold on merging.

Reply:
ok done